### PR TITLE
Unchained Keeps: Move chain.BondedECDSAKeep to discrete chain.BondedECDSAKeepHandle

### PR DESCRIPTION
**Pull request by Shadowfiend**
This PR converts `chain.BondedECDSAKeep`, which was an embedded
interface whose methods were all parametrized by a particular keep's
address, as a `common.Address`, to the `chain.BondedECDSAKeepHandle`
interface. This new interface represents a handle to a discrete underlying
keep, and its methods are now considered to be attached to the handle's
underlying keep. The `GetKeepAtIndex` method on
`chain.BondedECDSAKeepFactory`, which is still embedded in `chain.Handle`,
now returns a `chain.BondedECDSAKeep` instead of a `common.Address`.
A new function, `GetKeepWithID`, is also now available to look up a handle
given a keep's address.

This change cascades through a lot of files, and in general most places where
a keep's address was being passed now receive a handle to the keep instead.

As part of this PR, two small tidbits that were missed in #704 and #707 are
placed as the first two commits: client.go was still referencing `eth.` instead
of `chain.` after the rename of the `pkg/chain` external package name to `chain`,
and `pkg/local` still had keep-related functions in `local.go` instead of being
split out to a separate file. These now mirror the usage throughout the rest of
the codebase.

### CODE REVIEW

File: pkg/chain/chain.go
Comment (Shadowfiend) [reply pkg/chain/chain.go:109]: I believe that's how it currently works, and I'm comfortable with that. I can drop the TODO in the next PR---I think it's been there a while hehe.

File: pkg/client/client.go
Comment (pdyraga) [inline pkg/client/client.go:372]: I think we need to change `%v` to `%s` here now so that `String()` is called under the hood.

File: pkg/client/client.go
Comment (pdyraga) [inline pkg/client/client.go:120]: It doesn't seem `String()` is called even though we use `%s` here:

> 2021-03-02T12:30:47.315+0100	ERROR	keep-ecdsa	failed to verify if keep [�](Q����`����Y/�P�] is still active: [response [[]] was not long enough to interpret while resolving original error [no contract code at given address]]; subscriptions for keep signing and closing events are skipped

File: pkg/client/client.go
Comment (pdyraga) [inline pkg/client/client.go:111]: We use the same error message as for the `keep.IsActive()` check below. Can we differentiate them a little so that the node operator can be sure which call failed?

File: pkg/chain/chain.go
Comment (Shadowfiend) [reply pkg/chain/chain.go:109]: I think it's great though, I meant to ask about a couple of these TODOs. Have to clean them up when we notice them otherwise they stick around forever haha.

File: pkg/chain/celo/bonded_ecdsa_keep_handle.go
Comment (Shadowfiend) [reply pkg/chain/celo/bonded_ecdsa_keep_handle.go:149]: :grimacing: will fix.

File: pkg/client/client.go
Comment (Shadowfiend) [reply pkg/client/client.go:227]: In general it shouldn't matter. But, this and the bad log output you flagged above are both due to an issue with how go-ethereum handles formatting that was fixed in [v1.9.25](https://github.com/ethereum/go-ethereum/commit/bd848aad7c4e1f7d1eaecd9ea7ee23785090768a). We're trailing pretty hard, so I'm wondering if it's worth checking to see if a 1.9.25 bump would break everything, but I don't think the refactor should be blocked on it.

We're also about to swap all this out with the *ID types, which should handle stringification properly. But we're also merging this to master, so I'm a little torn about what to do heh.

Comment (pdyraga) [reply pkg/client/client.go:227]: Will we have the code`*ID` types ready soon? Because if so, we can maybe merge one PR right after another. 🤷

Comment (Shadowfiend) [reply pkg/client/client.go:227]: This week I think. I have a PR coming with the tBTC Application refactor, and then I think the IDs will be next.

File: pkg/chain/chain.go
Comment (Shadowfiend) [reply pkg/chain/chain.go]: Dropped.

File: pkg/chain/celo/bonded_ecdsa_keep_handle.go
Comment (Shadowfiend) [reply pkg/chain/celo/bonded_ecdsa_keep_handle.go]: Done.

File: pkg/client/client.go
Comment (Shadowfiend) [reply pkg/client/client.go]: Done.

**Comment by Shadowfiend** [comment]
This is ready for a follow-up review, provided we're comfortable with having some nasty log output until the ID changes land. Otherwise I'll go through and re-add `.String()` calls.

**Comment by pdyraga** [comment]
I am comfortable with merging it without `.String()` calls. Excellent work 👏

**Review by pdyraga — APPROVED** [review]

**Pull request merged**
